packages/beacon-node/src/chain/produceBlock/computeNewStateRoot.ts

```diff
@@ -41,5 +42,7 @@ export function computeNewStateRoot(
     metrics
   );
 
-  return postState.hashTreeRoot();
+  const proposerBalanceAfter = postState.balances.get(block.proposerIndex);
+
+  return {newStateRoot: postState.hashTreeRoot(), proposerReward: BigInt(proposerBalanceAfter - proposerBalanceBefore)};
```

Comment:
> so the consensus value to be passed is not the proposer balance reward, but total of reward. i think track individual reward penalty segments for : attestations, sync aggregate, proposer slashings, attester slashings as well as the actual proposer reward in epoch context as and when the block is processed and total them up
> 
> because the AIM is to make sure proposer makes the best block and not really about the proposer but indirectly proposer benefists because he/she gets some reward for including all of these things
> 
> so total would be = attestations + syncAggregate + proposerSlashings + attesterSlashings + the proposer award in this proposer award can't also be simply calculated as : poststate.balance - prestate.balance, because the proposer can also be part of some attestation or lets say whislleblower for slashings, so the balance diff will have those rewards added to it to
> 
> although proposer total diff is correlated to the block total reward, but i think thats not what is being expected in response here. could you check it with other teams

I've asked around, seems like the consensus value here = `total` in the block reward API, which is the total reward only for the proposer to include attestation, sync committee, slashings into the block. 

> proposer award can't also be simply calculated as : poststate.balance - prestate.balance
This is a good point. I have updated the calculation